cmd/influxd/server_single_integration_test.go

```diff
@@ -0,0 +1,264 @@
+package main_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb"
+	"github.com/influxdb/influxdb/influxql"
+
+	main "github.com/influxdb/influxdb/cmd/influxd"
+)
+
+func TestNewServer(t *testing.T) {
+	// Uncomment this to see the test fail when running for a second time in a row
+	t.Skip()
+
+	var (
+		join    = ""
+		version = "x.x"
+	)
+
+	tmpDir := os.TempDir()
+	tmpBrokerDir := filepath.Join(tmpDir, "broker")
+	tmpDataDir := filepath.Join(tmpDir, "data")
+	t.Logf("Using tmp directorie %q for broker\n", tmpBrokerDir)
+	t.Logf("Using tmp directorie %q for data\n", tmpDataDir)
+
+	c := main.NewConfig()
+	c.Broker.Dir = tmpBrokerDir
+	c.Data.Dir = tmpDataDir
+
+	now := time.Now()
+	var spinupTime time.Duration
+
+	s := main.Run(c, join, version)
+
+	defer func() {
+		s.Close()
+		t.Log("Shutting down server and cleaning up tmp directories")
+		err := os.RemoveAll(tmpBrokerDir)
+		if err != nil {
+			t.Logf("Failed to clean up %q: %s\n", tmpBrokerDir, err)
+		}
+		err = os.RemoveAll(tmpDataDir)
+		if err != nil {
+			t.Logf("Failed to clean up %q: %s\n", tmpDataDir, err)
+		}
+	}()
+
+	defer func() {
+		t.Log("Shutting down server and cleaning up tmp directories")
+		err := os.RemoveAll(tmpBrokerDir)
+		if err != nil {
+			t.Logf("Failed to clean up %q: %s\n", tmpBrokerDir, err)
+		}
+		err = os.RemoveAll(tmpDataDir)
+		if err != nil {
+			t.Logf("Failed to clean up %q: %s\n", tmpDataDir, err)
+		}
+	}()
+
+	ready := make(chan bool, 1)
+	go func() {
```

Comment:
Yes, basically this will spin up and start polling, and the following for loop watches the channels to get either the ready state or it times out.  If you have a better pattern than this I'm open to it.